common/log/logger.go

```diff
@@ -45,6 +45,8 @@ import (
 //  Note: msg should be static, it is not recommended to use fmt.Sprintf() for msg.
 //        Anything dynamic should be tagged.
 type Logger interface {
+	internalLogger
```

Comment:
This design choice is weird. Could we have a separate interface that loggerImpl implements?

Reply:
Addressed